docs/docs/slot-validation-actions.mdx

```diff
@@ -0,0 +1,103 @@
+in the call that the default action `action_extract_slots` makes to the action server.
+
+:::note
+Slots mapped to a form in the mapping `conditions` will be ignored by `ValidationAction` unless you override
+the `required_slots` method to include the slots you want to extract and / or validate.
+:::
+
+### Validation of Slots with Predefined Mappings
+
+To validate slots with a predefined mapping, you must write functions named `validate_<slot_name>`.
+
+In the following example, the value for slot `location` is capitalized only if the extracted value is of type string:
+
+```python
+from typing import Text, Any, Dict
+
+from rasa_sdk import Tracker, ValidationAction
+from rasa_sdk.executor import CollectingDispatcher
+from rasa_sdk.types import DomainDict
+
+
+class ValidatePredefinedSlots(ValidationAction):
+    def validate_location(
+        self,
+        slot_value: Any,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: DomainDict,
+    ) -> Dict[Text, Any]:
+        """Validate location value."""
+        if isinstance(slot_value, str):
+            # validation succeeded, capitalize the value of the "location" slot
+            return {"location": slot_value.capitalize()}
+        else:
+            # validation failed, set this slot to None
+            return {"location": None}
+```
+
+### Extraction of Custom Slot Mappings
+
+In order to define custom extraction code, you must write an `extract_<slot_name>` method for every slot with a
+custom slot mapping.
+
+The following example shows the implementation of a custom action that extracts the slot `count_of_insults` to keep
+track of the user's attitude.
+
+```python
+from typing import Dict, Text, Any
+
+from rasa_sdk import Tracker
+from rasa_sdk.executor import CollectingDispatcher
+from rasa_sdk.forms import ValidationAction
+
+
+class ValidateCustomSlotMappings(ValidationAction):
+    async def extract_count_of_insults(
+        self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict
+    ) -> Dict[Text, Any]:
+        intent_of_last_user_message = tracker.get_intent_of_latest_message()
+        current_count_of_insults = tracker.get_slot("count_of_insults")
+        if intent_of_last_user_message == "insult":
+           current_count_of_insults += 1
+
+        return {"count_of_insults": current_count_of_insults}
+```
+
+## `FormValidationAction` class
+
+If certain custom slot mappings should only be extracted and / or validated within the context of a form,
+make sure your custom action inherits from `FormValidationAction` rather than from `ValidationAction`.
+It is not required to use mapping conditions for custom slot mappings defined in a custom action extending
+`FormValidationAction`.
+Note that this custom action will only run if the form is activated.
+
+To learn more about how to implement this class, see Forms [Advanced Usage](./forms.mdx#advanced-usage).
```

Comment:
```suggestion
A `FormValidationAction` custom action will only run if the form it specifies in its name is activated.
If certain custom slot mappings should only be extracted and / or validated within the context of a form,
the custom action should inherit from `FormValidationAction` rather than from `ValidationAction`.
Since `FormValidationAction` runs on every user turn only as long as the form is active, it is not required to use mapping conditions for custom slot mappings defined in a custom action extending
`FormValidationAction`.

To learn more about how to implement this class, see Forms [Advanced Usage](./forms.mdx#advanced-usage).
```